### PR TITLE
Support naming convention

### PR DESCRIPTION
**Pull request by alfretoine**
## Self proof reading checklist

- [ ] Did I run Pronto and fixed all warnings
- [ ] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [ ] Did I test the right thing?
- [ ] Did I test corner cases, non happy path (defensive testing)?
- [ ] Check that I used the ad-hoc patterns and created files accordingly (Presenters, Services, Search object, Form object,...)
- [ ] Update documentation,readme accordingly
- [ ] Check code efficiency with record intensive project

Thanks!

### CODE REVIEW

File: app/models/state.rb [hunk 1]
```diff
@@ -47,4 +48,22 @@ def activity_level?
   def to_unified_h
     { name: name }
   end
+
+  def names(naming_patterns, activity, state)
```
Comment (mestachs) [inline app/models/state.rb]: strange to accept a state when you are call the method on a state

File: app/models/state.rb [hunk 2]
```diff
@@ -21,7 +22,7 @@ class State < ApplicationRecord
 
   validates :name, uniqueness: {
     scope:   :project_id,
-    message: "should be unique per project"
+    message: "state name should be unique per project"
```
Comment (mestachs) [inline app/models/state.rb]: we need a similar uniqueness validation for short_name

File: app/models/formula_mapping.rb [hunk 3]
```diff
@@ -23,4 +23,31 @@ class FormulaMapping < ApplicationRecord
...
+        state_code:          formula.code
+      }
+      {
+        long:  format(naming_patterns[:long], substitutions).strip,
+        short: format(naming_patterns[:short], substitutions).strip,
+        code:  format(naming_patterns[:code], substitutions).strip
+      }
+    else
+      name = formula.code.humanize.strip
+      {
```
Comment (mestachs) [inline app/models/formula_mapping.rb]: I would create a Struct or small data class
and replace name[:short]
   name.short 
   name.long
   name.code
allows to fails faster on type like 
    name[:short_name] 
with an explicit error message (no such method short_name) vs returning nil
